src/toil/cwl/cwltoil.py

```diff
@@ -423,7 +423,7 @@ def run(self, fileStore):
         os.environ["TMPDIR"] = os.path.realpath(opts.pop("tmpdir", None) or fileStore.getLocalTempDir())
         outdir = os.path.join(fileStore.getLocalTempDir(), "out")
         os.mkdir(outdir)
-        tmp_outdir_prefix = os.path.join(os.environ["TMPDIR"], "out_tmpdir")
+        tmp_outdir_prefix = os.path.join(opts.pop("workdir", None) or os.environ["TMPDIR"], "out_tmpdir")
```

Comment:
Could write this as `opts.pop("workdir", os.environ["TMPDIR"])` I think.

Reply:
Peter, thanks you can do but it becomes a problem if `workdir` is set but has a value of None (or False, or an empty string). I do this `or` bit only when I'm not sure the inputs won't be None so it's a little more safe.